packages/docusaurus/src/commands/deploy.ts

```diff
@@ -97,18 +97,23 @@ export default async function deploy(
     (projectName.indexOf('.github.io') !== -1 ? 'master' : 'gh-pages');
   console.log(`${chalk.cyan('deploymentBranch:')} ${deploymentBranch}`);
 
+  const useSSH = process.env.USE_SSH;
+
   const githubHost =
     process.env.GITHUB_HOST || siteConfig.githubHost || 'github.com';
+  const githubPort =
+    process.env.GITHUB_PORT || siteConfig.githubPort || (useSSH && useSSH.toLowerCase() === 'true'
+      ? '22'
+      : '433');
```

Comment:
do we absolutely need an explicit port in all cases? Isn't this implied by the protocol?

Reply:
It is not necessary. I did it in this way just to make it easier to implement.
Because if we doesn't specify the port we need to remove the character ':'.

Reply:
Done.